src/python_pybind11/src/Vector3.hh

```diff
@@ -0,0 +1,166 @@
+    .def("rounded", &Class::Rounded, "Get a rounded version of this vector")
+    .def("set", &Class::Set, "Set the contents of the vector")
+    .def("cross",
+         &Class::Cross,
+         "Return the cross product of this vector with another vector.")
+    .def("dot",
+         &Class::Dot,
+         "Return the dot product of this vector and another vector")
+    .def("abs", &Class::Abs,
+         "Return the absolute dot product of this vector and "
+         "another vector. This is similar to the Dot function, except the "
+         "absolute value of each component of the vector is used.")
+    .def("abs_dot", &Class::AbsDot, " Get the absolute value of the vector")
+    .def("perpendicular",
+         &Class::Perpendicular,
+         "Return a vector that is perpendicular to this one.")
+    .def("normal", &Class::Normal, "Get a normal vector to a triangle")
+    .def("dist_to_line",
+         &Class::DistToLine,
+         "Get distance to an infinite line defined by 2 points.")
+    .def("max",
+         py::overload_cast<const Class&>(&Class::Max),
+         "Set this vector's components to the maximum of itself and the "
+         "passed in vector")
+    .def("max", py::overload_cast<>(&Class::Max, py::const_),
+         "Get the maximum value in the vector")
+    .def("min", py::overload_cast<const Class&>(&Class::Min),
+         "Set this vector's components to the minimum of itself and the "
+         "passed in vector")
+    .def("min", py::overload_cast<>(&Class::Min, py::const_),
+         "Get the minimum value in the vector")
+    .def(py::self + py::self)
+    .def(py::self += py::self)
+    .def(py::self + T())
+    .def(py::self += T())
+    .def(py::self * py::self)
+    .def(py::self *= py::self)
+    .def(py::self * T())
+    .def(py::self *= T())
+    .def(py::self - py::self)
+    .def(py::self -= py::self)
+    .def(py::self - T())
+    .def(py::self -= T())
+    .def(py::self / py::self)
+    .def(py::self /= py::self)
+    .def(py::self / T())
+    .def(py::self /= T())
+    .def(py::self != py::self)
+    .def(py::self == py::self)
+    .def(-py::self)
+    .def("equal",
+         py::overload_cast<const Class&, const T&>(&Class::Equal, py::const_),
+         "Equality test with tolerance.")
+    .def("equal",
+         py::overload_cast<const Class&>(&Class::Equal, py::const_),
+         "Equal to operator")
+    .def("is_finite",
+         &Class::IsFinite,
+         "See if a point is finite (e.g., not nan)")
+    .def("correct", &Class::Correct, "Corrects any nan values")
+    .def("x", py::overload_cast<>(&Class::X), "Get the x value.")
+    .def("y", py::overload_cast<>(&Class::Y), "Get the y value.")
+    .def("z", py::overload_cast<>(&Class::Z), "Get the z value.")
+    .def("x", py::overload_cast<const T&>(&Class::X), "Get the x value.")
+    .def("y", py::overload_cast<const T&>(&Class::Y), "Get the y value.")
+    .def("z", py::overload_cast<const T&>(&Class::Z), "Get the z value.")
```

Comment:
I think this and the two above are setters, not getters. Please adjust the doc-string